curl/src/client.cc

```diff
@@ -261,6 +261,33 @@ namespace leatherman { namespace curl {
         if (result != CURLE_OK) {
             throw http_request_exception(ctx.req, curl_easy_strerror(result));
         }
+        result = curl_easy_setopt(_handle, CURLOPT_SEEKFUNCTION, seek_body);
+        if (result != CURLE_OK) {
+            throw http_request_exception(ctx.req, curl_easy_strerror(result));
+        }
+        result = curl_easy_setopt(_handle, CURLOPT_SEEKDATA, &ctx);
+        if (result != CURLE_OK) {
+            throw http_request_exception(ctx.req, curl_easy_strerror(result));
+        }
+
+        switch (method) {
```

Comment:
I'm thinking it over. These two methods are closely related, but the alternatives are that they have to be called in a certain order, or we merge them into one. I'm ok with it the way it is.